### PR TITLE
Controllers refactoring

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

File: Controller/ControllerAbstract.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+namespace FSi\Bundle\AdminBundle\Controller;
+
+use FSi\Bundle\AdminBundle\Admin\Context\ContextManager;
+use FSi\Bundle\AdminBundle\Admin\Element;
+use FSi\Bundle\AdminBundle\Exception\ContextException;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class ControllerAbstract
```
Comment (rn0) [inline Controller/ControllerAbstract.php]: missing `abstract class`

File: spec/FSi/Bundle/AdminBundle/Controller/BatchControllerSpec.php [hunk 2]
```diff
@@ -49,7 +51,7 @@ function it_throws_exception_when_context_does_not_return_response(
         $manager->createContext('fsi_admin_batch', $element)->willReturn($context);
         $context->handleRequest($request)->willReturn(null);
 
-        $this->shouldThrow(new ContextException("Context which handles batch action must return instance of \\Symfony\\Component\\HttpFoundation\\Response"))
+        $this->shouldThrow('FSi\Bundle\AdminBundle\Exception\ContextException')
```
Comment (bartekgruszka) [inline spec/FSi/Bundle/AdminBundle/Controller/BatchControllerSpec.php]: You should update also this second test for consistency, `$this->shouldThrow('Symfony\Component\HttpKernel\Exception\NotFoundHttpException')`

File: Controller/ControllerAbstract.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+     * @param \FSi\Bundle\AdminBundle\Admin\Element $element
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string $route
+     * @return null|\Symfony\Component\HttpFoundation\Response
+     */
+    protected function handleRequest(Element $element, Request $request, $route)
+    {
+        $context = $this->contextManager->createContext($route, $element);
+
+        if (!isset($context)) {
```
Comment (bartekgruszka) [inline Controller/ControllerAbstract.php]: Maybe it would be better to check `instanceof` $context

**Comment by bartekgruszka** [comment]
:+1:

**Comment by rn0** [comment]
:+1:

**Pull request merged**